### PR TITLE
Hdfs

### PR DESCRIPTION
**Pull request by fiorda**

### CODE REVIEW

File: pybossa/hdfs/client.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    def reinit(self, principal):
+        current_app.logger.info('running kinit')
+        try:
+            check_call(['kinit', '-kt', self.keytab, principal])
+        except CalledProcessError as e:
+            current_app.logger.exception(e.output)
+            raise
+
+    def get_ticket(self):
```
Comment (jayflo) [inline pybossa/hdfs/client.py]: What is a ticket

Comment (jayflo) [reply pybossa/hdfs/client.py]: https://www.roguelynn.com/words/explain-like-im-5-kerberos/

File: pybossa/view/fileproxy.py [hunk 2]
```diff
@@ -93,11 +107,53 @@ def encrypted_file(store, bucket, project_id, path):
...
+    payload = signer.loads(signature, max_age=timeout)
+    task_id = payload['task_id']
+    check_allowed(current_user.id, task_id, project, request.path)
+
+    client = HDFSKerberos(**current_app.config['HDFS_CONFIG'][cluster])
+    try:
+        content = client.get('/{}'.format(path))
+        project_encryption = project['info'].get('ext_config', {}).get('encryption', {})
+        if project_encryption:
+            secret = get_secret_from_vault(project_encryption)
```
Comment (jayflo) [inline pybossa/view/fileproxy.py]: So in this case, each project will have it's own secret for decrypting (which is different from the bcos files).  What is that vault?

File: pybossa/view/fileproxy.py [hunk 3]
```diff
@@ -93,11 +107,53 @@ def encrypted_file(store, bucket, project_id, path):
...
     response = Response(decrypted, content_type=_key.content_type)
-
     response.headers.add('Content-Encoding', _key.content_encoding)
     response.headers.add('Content-Disposition', _key.content_disposition)
-    response.headers.add('Cache-Control', 'no-store')
-    response.headers.add('Pragma', 'no-cache')
     return response
+
+
+@blueprint.route('/hdfs/<string:cluster>/<int:project_id>/<path:path>')
```
Comment (jayflo) [inline pybossa/view/fileproxy.py]: I'm guessing this route wouldn't be used for the public instance.  Do we want to somehow disable routes like this in that situation?

File: pybossa/hdfs/client.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        principal = self.proxy or self.user
```
[the remark this answers is not in the document]
Comment (fiorda) [reply pybossa/hdfs/client.py]: probably I refactored at some point and this was left in the wrong spot

File: pybossa/view/fileproxy.py (hunk 3 above)
Comment (fiorda) [reply pybossa/view/fileproxy.py]: we can

File: pybossa/hdfs/client.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
+from subprocess import check_call, CalledProcessError
+
+from flask import current_app
+from hdfs.ext.kerberos import KerberosClient
+from requests import Session
```
Comment (dchhabda) [inline pybossa/hdfs/client.py]: this line can be removed

File: test/test_view/test_fileproxy.py [hunk 6]
```diff
@@ -208,3 +208,146 @@ def test_proxy_key_not_found(self, create_connection):
...
+
+        with patch.dict(self.flask_app.config, self.app_config):
+            res = self.app.get(req_url, follow_redirects=True)
+            assert res.status_code == 200, res.status_code
+            assert res.data == 'the content', res.data
+
+    @with_context
+    @patch('pybossa.view.fileproxy.HDFSKerberos.get')
+    @patch('pybossa.view.fileproxy.requests.get')
+    def test_proxy_key_err(self, http_get, hdfs_get):
```
Comment (dchhabda) [inline test/test_view/test_fileproxy.py]: hdfs_get unused here?

Comment (fiorda) [reply test/test_view/test_fileproxy.py]: it is used to avoid making the actual request

**Pull request merged**